keel-igor/src/main/kotlin/com/netflix/spinnaker/keel/igor/BuildLifecycleMonitor.kt

```diff
@@ -43,56 +47,53 @@ class BuildLifecycleMonitor(
 
   override suspend fun monitor(task: MonitoredTask) {
     val buildData = parseAndValidate(task) ?: return
-    kotlin.runCatching {
+    runCatching {
       artifactMetadataService.getArtifactMetadata(buildData.buildNumber, buildData.commitId)
     }.onSuccess { metadata ->
       val buildMetadata = metadata?.buildMetadata
       if (buildMetadata == null) {
         log.error("Error fetching status for $task, response was null")
-        handleFailureFetching(task)
+        handleFailureFetchingStatus(task)
       } else {
         when (buildMetadata.status) {
           "BUILDING" -> publishRunningEvent(task)
           "SUCCESS" -> publishSucceededEvent(task, buildMetadata)
           "FAILURE" -> publishFailedEvent(task, buildMetadata)
           "ABORTED" -> publishAbortedEvent(task, buildMetadata)
+          // UNSTABLE means build passed but tests failed (might need to reevaluate status in the future)
```

Comment:
The official definition is: 
> The Build had some errors but they were not fatal. A Build is unstable if it was built successfully and one or more publishers report it unstable. For example if the JUnit publisher is configured and a test fails then the Build will be marked unstable.

Source, if you want to add the link to the comment: https://www.jenkins.io/doc/book/glossary/